src/platform/plugins/shared/es_ui_shared/static/forms/hook_form_lib/components/form_data_provider.test.tsx

```diff
@@ -104,41 +95,27 @@ describe('<FormDataProvider />', () => {
       );
     };
 
-    const setup = registerTestBed(TestComp, {
-      memoryRouter: { wrapComponent: false },
-    });
-
-    const {
-      form: { setInputValue },
-      find,
-    } = setup() as TestBed;
+    render(<TestComp />);
 
     expect(onFormData).toBeCalledTimes(0); // Not present in the DOM yet
 
     // Make some changes to the form fields
-    await act(async () => {
-      setInputValue('nameField', 'updated value');
-    });
+    const nameField = screen.getByTestId('nameField');
+    await user.clear(nameField);
+    await user.type(nameField, 'updated value');
```

Comment:
nit: I think we can simplify it by:

```suggestion
    await user.type(nameField, 'updated value', { selectAll: true });
```

there are few other places in this file (and others!) that I think we could do the same